internal/git.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2021 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package internal
+
+import (
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+	giturls "github.com/whilp/git-urls"
+)
+
+// SetGitConfigContext attempts to set the org and repo
+// based on the .git/ directory, provided the user has
+// the config flag of no-git set to true.
+func SetGitConfigContext(c *cli.Context) {
+	// check to see if config and command allow for
+	// automatic setting of org and repo.
+	if c.String(FlagNoGit) == "false" &&
+		c.String(FlagOrg) == "" &&
+		c.String(FlagRepo) == "" {
+		// set the org
+		logrus.Trace("attempting to set org from .git/")
+		err := c.Set(FlagOrg, GetGitConfigOrg("./"))
+
+		if err != nil {
+			logrus.Debug("failed to set org in context")
+		}
+
+		// set the repo
+		logrus.Trace("attempting to set repo from .git/")
+		err = c.Set(FlagRepo, GetGitConfigRepo("./"))
+
+		if err != nil {
+			logrus.Debug("failed to set repo in context")
+		}
+	}
+}
+
+// GetGitConfigOrg opens the git repository, fetches
+// the remote origin url, and parses the url to find
+// the org of the current working directory.
+func GetGitConfigOrg(path string) string {
+	// open repository
+	r, err := git.PlainOpen(path)
+
+	// on failure, return empty string to allow for
+	// potential manual setting of org to process
+	if err != nil {
+		return ""
+	}
+
+	// fetch remote origin
+	origin, err := r.Remote("origin")
+	if err != nil {
+		return ""
+	}
+	url, err := giturls.Parse(origin.Config().URLs[0])
+	if err != nil {
+		return ""
+	}
+
+	// path is :org/:repo.git - get org
+	org := strings.Split(url.Path, "/")[0]
```

Comment:
Should add some extra validation to ensure you catch the scenario where `url` has no `/`. I would typically do something like the following:

```golang
splitOrg := strings.Split(url.Path, "/")
if len(splitOrg) != 1 {
   return fmt.Errorf("unexpected url of %s", url.Path)
}
```

Utilizing [SplitN](https://pkg.go.dev/strings#SplitN) is also likely preferable since you dictate how many parts you expect to be returned.